src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ConstructedEETypeNode.cs

```diff
@@ -26,6 +26,14 @@ public override bool ShouldSkipEmittingObjectNode(NodeFactory factory)
             return false;
         }
 
+        public override bool InterestingForDynamicDependencyAnalysis
+        {
+            get
+            {
+                return _type.IsDefType && _type.HasGenericVirtualMethod();
```

Comment:
The base type, EETypeNode, explicitly has Debug.Assert(!type.IsRuntimeDeterminedSubtype);